### PR TITLE
CORE-10444 Initial registration implementation

### PR DESCRIPTION
**Pull request by ac101m**
# Changes
- Add avro schema for interop alias identity data

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropAliasIdentity.avsc [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      "name": "groupId",
+      "type": "string"
+    },
+    {
+      "doc": "My alias x500 name.",
+      "name": "aliasX500Name",
+      "type": "string"
+    },
+    {
+      "doc": "Holding Identity of Virtual Node hosting.",
```
Comment (szymonsztuka) [inline data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropAliasIdentity.avsc]: The description is confusing, especially comparing to a field name  "hostingVnode".
I think this field is redundant, as the short hash of Vnode will be a key for this record.

File: data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropAliasIdentity.avsc [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  "name": "InteropAliasIdentity",
+  "namespace": "net.corda.data.interop",
+  "fields": [
+    {
+      "doc": "UUID of interop group.",
+      "name": "groupId",
+      "type": "string"
+    },
+    {
+      "doc": "My alias x500 name.",
```
Comment (szymonsztuka) [inline data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropAliasIdentity.avsc]: Should be: "Interop Identity X500Name"

File: data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropAliasIdentity.avsc [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "type": "record",
+  "name": "InteropAliasIdentity",
+  "namespace": "net.corda.data.interop",
+  "fields": [
+    {
+      "doc": "UUID of interop group.",
```
Comment (szymonsztuka) [inline data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropAliasIdentity.avsc]: Should be: "Interop Group Name"  or  "Interop Group UUID".

File: data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropAliasIdentity.avsc [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      "name": "groupId",
+      "type": "string"
```
Comment (szymonsztuka) [inline data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropAliasIdentity.avsc]: Maybe be UUID field, can Avro handle UUID.

File: data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropAliasIdentity.avsc [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{
+  "type": "record",
+  "name": "InteropAliasIdentity",
```
Comment (ac101m) [inline data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropAliasIdentity.avsc]: Needs to be renamed to just `InteropIdentity`

File: data/topic-schema/src/main/java/net/corda/schema/Schemas.java [hunk 6]
```diff
@@ -106,6 +106,8 @@ private Flow() {
         public static final String FLOW_INTEROP_EVENT_TOPIC = "flow.interop.event";
         public static final String FLOW_INTEROP_EVENT_STATE_TOPIC = getStateAndEventStateTopic(FLOW_INTEROP_EVENT_TOPIC);
         public static final String FLOW_INTEROP_EVENT_DLQ_TOPIC = getDLQTopic(FLOW_INTEROP_EVENT_TOPIC);
+
+        public static final String INTEROP_ALIAS_IDENTITY_TOPIC = "interop.alias.identity";
```
Comment (ac101m) [inline data/topic-schema/src/main/java/net/corda/schema/Schemas.java]: Needs to be renamed to `INTEROP_IDENTITY_TOPIC`

File: data/topic-schema/src/main/resources/net/corda/schema/Flow.yaml [hunk 7]
```diff
@@ -85,6 +85,20 @@ topics:
     producers:
       - interop
     config:
+  InteropIdentitiesTopic:
+    name: interop.alias.identity
```
Comment (ac101m) [inline data/topic-schema/src/main/resources/net/corda/schema/Flow.yaml]: Remove alias

File: gradle.properties [hunk 8]
```diff
@@ -12,8 +12,8 @@ cordaProductVersion = 5.1.0-INTEROP
 ##   a per module property in which case module versions can change independently.
 ## IMPORTANT:
 ## The interop feature branches track api revisions separately to the mainline branch.
-## API version of last merge from corda mainline: 5
-cordaApiRevision = 9
+## API version of last merge from corda mainline: 2
```
Comment (ac101m) [inline gradle.properties]: Why did this go backwards?

Comment (szymonsztuka) [reply gradle.properties]: I guess maybe it was lost during a merge etc?

**Review by szymonsztuka — APPROVED** [review]

**Review by jmacmahonr3 — APPROVED** [review]
LGTM

File: data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropIdentity.avsc [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    {
+      "doc": "Interop identity x500 name.",
+      "name": "x500Name",
+      "type": "string"
+    },
+    {
+      "doc": "Holding Identity of hosting Virtual Node.",
+      "name": "hostingVnode",
+      "type": "string"
+    }
```
Comment (ac101m) [inline data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropIdentity.avsc]: @jmacmahonr3 This is the only part I think needs another look.

Comment (jmacmahonr3) [reply data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropIdentity.avsc]: Ah yeah ... 
I don't think it's a blocking issue ... 
We can update the names as we go

Comment (szymonsztuka) [reply data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropIdentity.avsc]: The doc and name are contadicting: "Holding Identity of... " vs "hostingVnode", so what is actually in this field? Is there a Holding identity or Vnode short hash?

**Review by szymonsztuka — CHANGES_REQUESTED** [review]

File: data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropIdentity.avsc [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  "type": "record",
+  "name": "InteropIdentity",
+  "namespace": "net.corda.data.interop",
+  "fields": [
+    {
+      "doc": "Interop group ID.",
+      "name": "groupId",
+      "type": {
+        "type": "string",
+        "logicalType": "uuid"
```
Comment (szymonsztuka) [inline data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropIdentity.avsc:11]: Nice!

**Review by szymonsztuka — APPROVED** [review]

**Review by yarivus — COMMENTED** [review]
thx @corda/ledgerops

**Pull request merged**